service/src/main/java/bio/terra/pipelines/service/PipelinesService.java

```diff
@@ -195,17 +202,62 @@ public Map<String, Object> constructInputs(
         getServiceProvidedInputDefinitions(pipelineName);
 
     // add default values for service-provided inputs to the allPipelineInputs map
-    serviceProvidedInputDefinitions.stream()
-        .forEach(
-            inputDefinition -> {
-              String inputName = inputDefinition.getName();
-              Object inputValue = inputDefinition.getDefaultValue();
-              allPipelineInputs.put(
-                  inputName, inputValue); // store the string value; will cast later
-            });
+    serviceProvidedInputDefinitions.forEach(
+        inputDefinition -> {
+          String inputName = inputDefinition.getName();
+          Object inputValue = inputDefinition.getDefaultValue();
+          allPipelineInputs.put(inputName, inputValue); // store the string value; will cast later
+        });
 
     logger.info("All pipeline inputs: {}", allPipelineInputs);
 
     return allPipelineInputs;
   }
+
+  /**
+   * Prepare a list of CBAS WorkflowInputDefinitions using RecordLookup (i.e. reading from WDS) for
+   * a given pipeline and WDL method name.
+   *
+   * @param pipelineInputDefinitions
+   * @param wdlMethodName
+   * @return
+   */
+  public List<WorkflowInputDefinition> prepareCbasWorkflowInputRecordLookupDefinitions(
+      List<PipelineInputDefinition> pipelineInputDefinitions, String wdlMethodName) {
+
+    return pipelineInputDefinitions.stream()
+        .map(
+            pipelineInputDefinition -> {
+              String inputName = pipelineInputDefinition.getName();
+              ParameterTypeDefinition parameterTypeDefinition =
+                  mapInputTypeToCbasParameterType(pipelineInputDefinition.getType());
+              return new WorkflowInputDefinition()
+                  .inputName("%s.%s".formatted(wdlMethodName, inputName))
+                  .inputType(parameterTypeDefinition)
+                  .source(
+                      new ParameterDefinitionRecordLookup()
+                          .recordAttribute(inputName)
+                          .type(ParameterDefinition.TypeEnum.RECORD_LOOKUP));
+            })
+        .toList();
+  }
+
+  protected ParameterTypeDefinition mapInputTypeToCbasParameterType(String type) {
```

Comment:
alternatively could add a method in the PipelineInputTypeEnum to return the appropriate ParameterTypeDefinition, but this seems more straightforward